src/AutoRest.CSharp/LowLevel/Generation/LowLevelClientWriter.cs

```diff
@@ -262,38 +262,34 @@ private void WriteSubClientInternalConstructor(ConstructorSignature signature)
             _writer.Line();
         }
 
-        private void DeclareMethodParameter(CodeWriter writer, ConvenienceMethod convenienceMethod)
+        private void WriteSpreadBodyConverter(CodeWriter writer, ConvenienceMethod convenienceMethod, FormattableString? spreadVariable)
```

Comment:
The writer should have no idea what a spread is.  The writer should simply get a list of things to write, this feels like we need to move some logic back into the output types OR perhaps even input if this information can be known at translation time between cadl and cadl.json

Reply:
I believe we kind of have to have this somewhere because "convert the spread parameter back into a body model" is not always done by a one-line command (especially when there are arrays in them, you could find one inside our test project), it is actually a block of code.
We could consider to add a delegate inside `ConvenienceMethod` data record so that it could know how to write this thing, and the writer could call it if there is one, in this way, the writer does not have to know about anything about "spread parameters". Is this OK?

Reply:
And we should not put this inside the emitter, because eventually we still need to write the part about "constructing body model back" (this way) or "adhoc serialize the body parameter from the spread parameters"
In addition, if we put the "spread parameter logic" in emitter, it will force us to put the logic related with unions (spread into overload of convenience method) also in the emitter which makes the code in emitter a lot complicated than it is right now, and we agree that we should not do that to make the emitter complicated.